equistore-torch/include/equistore/torch/labels.hpp

```diff
@@ -76,24 +89,51 @@ class EQUISTORE_TORCH_EXPORT LabelsHolder: public torch::CustomClassHolder {
     /// the entry is not part of these labels.
     ///
     /// @param entry one of:
+    ///    - a `LabelsEntry`
     ///    - a 1-D torch::Tensor containing integers;
     ///    - a list of integers;
     ///    - a tuple of integers;
     torch::optional<int64_t> position(torch::IValue entry) const;
 
+    /// Print the names and values of these Labels to a string, including at
+    /// most `max_entries` entries (set this to -1 to print all entries), and
+    /// indenting all lines after the first with `indent` spaces.
+    std::string print(int64_t max_entries, int64_t indent) const;
+
+    /// Implementation of __str__ for Python
+    std::string __str__() const;
+
+    /// Implementation of __repr__ for Python
+    std::string __repr__() const;
+
     /// Get the underlying equistore::Labels
-    const equistore::Labels& as_equistore() const {
-        return labels_;
+    const equistore::Labels& as_equistore() const;
+
+    /// Is this a LabelsView or a Labels?
+    bool is_view() const {
+        return !labels_.has_value();
     }
+
+    /// Transform a view of Labels into owned Labels, which can be further given
+    /// to equistore functions. This does nothing if the Labels are already
+    /// owned.
+    LabelsHolder to_owned() const;
+
 private:
+    struct CreateView {};
```

Comment:
what is this struct `CreateView` doing? Could not find any usage of this

Reply:
leftover from development! I'll remove it

Reply:
Actually, this is used in the constructor just below, to differentiate it from the main one. I've added a comment about that.